### PR TITLE
[itests] multi hop test move

### PR DESCRIPTION
**Pull request by halseth**
We move multi hop tests to own files. In addition we make the basic funding and force close tests run through a list of commitment type.

This is preparation for having these tests being run for the anchor commitment type (#4001).

### CODE REVIEW

File: lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go [hunk 1]
```diff
@@ -17,6 +19,69 @@ import (
 	"github.com/lightningnetwork/lnd/lntypes"
 )
 
+func testMultiHopHtlcClaims(net *lntest.NetworkHarness, t *harnessTest) {
```
Comment (joostjager) [inline lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go]: I think this deserves its own file?

File: lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go [hunk 2]
```diff
@@ -17,6 +19,69 @@ import (
...
+			// bob: outgoing their commit watch and see, they sweep on chain
+			// bob: incoming their commit watch and learn preimage
+			// carol: incoming our commit know preimage
+			name: "remote chain claim",
+			test: testMultiHopHtlcRemoteChainClaim,
+		},
+	}
+
+	for _, subTest := range subTests {
+
```
Comment (joostjager) [inline lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go]: newline

File: lntest/itest/lnd_test.go [hunk 3]
```diff
@@ -1053,14 +1053,18 @@ func basicChannelFundingTest(t *harnessTest, net *lntest.NetworkHarness,
 		return nil, nil, nil, fmt.Errorf("unable to get bobs's "+
 			"balance: %v", err)
 	}
-	if aliceBal.Balance != int64(chanAmt-pushAmt-calcStaticFee(0)) {
+
+	aliceBalance := btcutil.Amount(aliceBal.Balance)
+	if aliceBalance != chanAmt-pushAmt-calcStaticFee(0) {
```
Comment (joostjager) [inline lntest/itest/lnd_test.go]: store `chanAmt-pushAmt-calcStaticFee(0)` in var?

File: lntest/itest/lnd_test.go [hunk 4]
```diff
@@ -2982,22 +2982,54 @@ func padCLTV(cltv uint32) uint32 {
 //
 // TODO(roasbeef): also add an unsettled HTLC before force closing.
 func testChannelForceClosure(net *lntest.NetworkHarness, t *harnessTest) {
-	success := t.t.Run("channelForceClosure", func(t *testing.T) {
-		ht := newHarnessTest(t, net)
+	// We'll test the scenario for all commitment types, to ensure outputs
+	// can be swept.
+	allTypes := []commitType{
+		commitTypeLegacy,
+		commitTypeTweakless,
```
Comment (joostjager) [inline lntest/itest/lnd_test.go]: So the main change in this commit is that we are now also running the itest for tweakless?

Comment (joostjager) [reply lntest/itest/lnd_test.go]: Maybe we don't want to do that, to keep the travis run time < 50 mins, if we aren't testing tweakless now either. Or just test tweakless and not legacy.

Comment (joostjager) [reply lntest/itest/lnd_test.go]: Can confirm that tests run < 50 mins when leaving out one commit type.

File: lntest/itest/lnd_test.go [hunk 5]
```diff
@@ -2982,6 +2982,28 @@ func padCLTV(cltv uint32) uint32 {
 //
 // TODO(roasbeef): also add an unsettled HTLC before force closing.
 func testChannelForceClosure(net *lntest.NetworkHarness, t *harnessTest) {
+	success := t.t.Run("channelForceClosure", func(t *testing.T) {
+		ht := newHarnessTest(t, net)
+
+		// Since we'd like to test failure scenarios with outstanding
+		// htlcs, we'll introduce another node into our test network:
+		// Carol.
+		carol, err := net.NewNode("Carol", []string{"--hodl.exit-settle"})
```
[the remark this answers is not in the document]
Comment (halseth) [reply lntest/itest/lnd_test.go]: to later specify anchors for this node

File: lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go (hunk 1 above)
Comment (halseth) [reply lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go]: done

File: lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go [hunk 6]
```diff
@@ -498,3 +498,76 @@ func checkPaymentStatus(ctxt context.Context, node *lntest.HarnessNode,
 
 	return nil
 }
+
+func createThreeHopNetwork(t *harnessTest, net *lntest.NetworkHarness,
```
[the remark this answers is not in the document]
Comment (halseth) [reply lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go]: done in last commit

File: lntest/itest/lnd_test.go [hunk 7]
```diff
@@ -2982,6 +2982,28 @@ func padCLTV(cltv uint32) uint32 {
 //
 // TODO(roasbeef): also add an unsettled HTLC before force closing.
 func testChannelForceClosure(net *lntest.NetworkHarness, t *harnessTest) {
+	success := t.t.Run("channelForceClosure", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (halseth) [reply lntest/itest/lnd_test.go]: fixed

File: lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go (hunk 2 above)
Comment (halseth) [reply lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go]: fixed

File: lntest/itest/lnd_test.go (hunk 3 above)
Comment (halseth) [reply lntest/itest/lnd_test.go]: done

File: lntest/itest/lnd_test.go (hunk 4 above)
Comment (halseth) [reply lntest/itest/lnd_test.go]: removed tweakless

File: lntest/itest/lnd_multi-hop_test.go [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/invoicesrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lntypes"
+)
+
+func testMultiHopHtlcClaims(net *lntest.NetworkHarness, t *harnessTest) {
+
```
Comment (guggero) [inline lntest/itest/lnd_multi-hop_test.go]: nit: extra newline.

**Pull request merged**